src/App.js

```diff
@@ -1,25 +1,64 @@
-import logo from './logo.svg';
-import './App.css';
+import React from 'react';
+import axios from 'axios';
+import { Container, Form, Button} from 'react-bootstrap'
+import Map from './map';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      location: {},
+      searchQuery: '',
+      imgSrc: '',
+      displayResults: false,
+    }
+  }
+  getLocationInfo = async (e) => {
+    e.preventDefault();
+    const url = `https://us1.locationiq.com/v1/search.php?key=${process.env.REACT_APP_CITY_KEY}&q=${this.state.searchQuery}&format=json`;
+    try {
+      const location = await axios.get(url);
+      const locationArray = location.data;
+      this.setState({
+        location: locationArray[0],
+        displayResults: true,
+        imgSrc: `https://maps.locationiq.com/v3/staticmap?key=${process.env.REACT_APP_CITY_KEY}&center=${locationArray[0].lat},${locationArray[0].lon}&zoom=13`,
+        lon: locationArray[0].lon,
+        lat: locationArray[0].lat
+      });
+    } catch (error) {
+      console.log(`😱 Axios request failed: ${error}`);
+    }
+  }
+  render() {
+    return (
+      <>
+            <Container>
+              <Form onSubmit={this.getLocationInfo} >
+                <Form.Group controlId="">
+                  <Form.Control type="text" placeholder="search for a city here" onChange={(e) => this.setState({ searchQuery: e.target.value })} />
```

Comment:
Nice work on this so far!

Feedback here: Remove setState call as this can cause some funky behavior (infinite loops and making component less maintainable). Instead create a method to handle event above render and pass it to onChange.